src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,389 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using static Microsoft.DotNet.Interactive.CSharpProject.BuildCacheFileUtilities;
+
+namespace Microsoft.DotNet.Interactive.CSharpProject;
+
+public static class RoslynWorkspaceUtilities
+{
+    public class ProjectItem
+    {
+        public string ItemSpec { get; }
+        public IReadOnlyDictionary<string, string> Metadata { get; }
+
+        internal ProjectItem()
+        {
+//            ItemSpec = taskItem.ItemSpec;
+//            Metadata = taskItem.MetadataNames.Cast<string>().ToDictionary(x => x, x => taskItem.GetMetadata(x));
+        }
+    }
+
+    public class BuildDataResults
```

Comment:
These probably shouldn't be nested classes and by convention they should each be in separate files.

Reply:
Done.

Did a refactor to move this class to its own file